pkgs/by-name/au/autosubst2/package.nix

```diff
@@ -0,0 +1,49 @@
+{
+  lib,
+  haskellPackages,
+  fetchFromGitHub,
+}:
+
+haskellPackages.mkDerivation {
+  pname = "autosubst2";
+  version = "unstable-2022-07-04";
+
+  src = fetchFromGitHub {
+    owner = "uds-psl";
+    repo = "autosubst2";
+    rev = "8a71e1dc4dea81f13a9572ea302064eb374566c6";
+    hash = "sha256-3n87NBi3NbuDb44/oEnzjNk/TAHUwATJYuaw70k/tpk=";
+  };
+  patches = [ ./imports.patch ];
+
+  isLibrary = true;
+  isExecutable = true;
+  libraryHaskellDepends = with haskellPackages; [
+    base
+    parsec
+    containers
+    mtl
+    array
+    wl-pprint
+    graphviz
+    text
+  ];
+  executableHaskellDepends = with haskellPackages; [
+    base
+    optparse-applicative
+    wl-pprint
+    directory
+  ];
+  testHaskellDepends = with haskellPackages; [
+    base
+    QuickCheck
+    containers
+  ];
+  doCheck = false;
```

Comment:
Explanation for why checking is disabled would be nice — are there no tests/failing tests/testing can't fundamentally work/other reasons?

Reply:
That is because the upstream doesn't have a `Spec.hs` file but it has an useless `as2-test`. I think `haskellPackages.mkDerivation` uses it anyway if it is declared.

https://github.com/uds-psl/autosubst2/blob/8a71e1dc4dea81f13a9572ea302064eb374566c6/as2.cabal#L55